packages/experimental/Button/SPEC.md

```diff
@@ -0,0 +1,190 @@
+- appearance="transparent": The `Button` is styled such that it has no background or border styling and is just emphasized through its content styling.
+
+### Icon
+
+The `Button` component can include an `icon` that appears before or after its `children`. If an `icon` is provided without any `children`, then the `Button` becomes an icon-only `Button`.
+
+### Shape
+
+- shape="rounded": The button as rounded corners. This is the default is shape is not set.
+- shape="circular": The button has completely round corners. A button of equal width and height will be a circle.
+- shape="square": The button has right-angle corners.
+
+### Sizes
+
+The `Button` component supports different sizing with at least three different sizes: `small`, `medium` (default) and `large`.
+
+### Block
+
+The `Button` component can completely fill the width of its container.
+
+### Loading
+
+The `Button` component can be loading if it's waiting for another action to occur before allowing itself to be interacted with.
+
+## API
+
+### Slots
+
+- `root` - The outer container representing the `Button` itself that wraps everything passed via the `children` prop.
+- `icon` - If specified, renders an `icon` either before or after the `children` as specified by the `iconPosition` prop.
+- `loader` - If specified, renders a `loader` before the `icon` and `children` while the `loading` flag is set to `true`.
+
+### Props
+
+```ts
+export type ButtonProps = Omit<IWithPressableOptions<ViewProps>, 'onPress'> & {
+  /**
+   * A button can have its content and borders styled for greater emphasis or to be subtle.
+   * - 'primary': Emphasizes the button as a primary action.
+   * - 'outline': Removes background styling.
+   * - 'subtle': Minimzes emphasis to blend into the background until hovered or focused.
+   * - 'transparent': Removes background and border styling.
+   */
+  appearance?: 'primary' | 'outline' | 'subtle' | 'transparent';
+
+  /**
+   * Icon slot that, if specified, renders an icon either before or after the `children` as specified by the
+   * `iconPosition` prop.
+   */
+  icon?: IconSourcesType;
+
+  // /**
+  //  * Loader slot that, if specified, renders a `loader` before the `icon` and `children` while the `loading` flag
+  //  * is set to `true`.
+  //  */
+  // loader?: ShorthandPropsCompat<React.HTMLAttributes<HTMLSpanElement>>;
+
+  /**
+   * A button can fill the width of its container.
+   * @default false
+   */
+  block?: boolean;
+
+  /**
+   * A button can show that it cannot be interacted with.
+   * @default false
+   */
+  disabled?: boolean;
+
+  /**
+   * A button can format its icon to appear before or after its content.
+   * @default 'before'
+   */
+  iconPosition?: 'before' | 'after';
+
+  // /**
+  //  * A button can show a loading indicator if it is waiting for another action to happen before allowing itself to
+  //  * be interacted with.
+  //  * @default false
+  //  */
+  // loading?: boolean;
```

Comment:
I think loading prop should be uncommented

Reply:
Yeah, we can uncomment these. I think I left them commented since we weren't sure if we were doing it yet.